BRNN.lua

```diff
@@ -0,0 +1,88 @@
+------------------------------------------------------------------------
+--[[ BRNN ]] --
+-- Encapsulates a forward, backward and merge module.
+-- Input is a tensor e.g batch x time x inputdim.
+-- Output is a tensor of the same length e.g batch x time x outputdim.
+-- Applies a forward rnn to input tensor in forward order
+-- and applies a backward rnn in reverse order.
+-- Reversal of the sequence happens on the chosen dim (defaults to 2).
+-- For each step, the outputs of both rnn are merged together using
+-- the merge module (defaults to nn.CAddTable() which sums the activations).
+------------------------------------------------------------------------
+
+local BRNN, parent = torch.class('nn.BRNN', 'nn.Container')
```

Comment:
I assume that inheriting from nn.Container will automatically give support for `clearState` and `parameters`?